lib/core/storage.js

```diff
@@ -179,9 +179,31 @@ export default class Storage {
     if (isUnset(value)) {
       Cookies.remove(_key, _options)
     } else {
-      Cookies.set(_key, value, _options)
-    }
 
+      // Support server set cookies
+      if (process.server) {
+        let setCookieHeader = this.ctx.res.getHeader('Set-Cookie')
+        if (!setCookieHeader) {
+          setCookieHeader = []
+        }
+        const existingCookieIndex = setCookieHeader.findIndex(element => element.split('=')[0] === _key)
```

Comment:
I would feel more confident using dedicated package for parsing cookies like `cookie`. Also it would make code a lot cleaner.

See example in `nuxt-i18n` module on handling server side cookie using that module:
https://github.com/nuxt-community/nuxt:
 - set: i18n/blob/ca9873b85f0d1e69763cafca01a7c7df51893107/src/plugins/main.js#L63,L74
 - read: https://github.com/nuxt-community/nuxt-i18n/blob/ca9873b85f0d1e69763cafca01a7c7df51893107/src/templates/middleware.js#L43,L48

Reply:
Totally agree! Will fix it :)

Reply:
Agree with @rchl. `cookie` has ~1KB overhead but seems safer. We can require it inside `if (process.server)` to prevent including in client-side bundle. Please ping me when you refactored to get merged ASAP.

Reply:
Actually this file even imports those packages already:

```js
import Cookies from 'js-cookie'
...
import { parse as parseCookie } from 'cookie'
```